lib/cocoapods/resolver.rb

```diff
@@ -380,6 +381,12 @@ def handle_resolver_error(error)
               'version requirement to your Podfile ' \
               "(e.g. `pod '#{name}', '#{lockfile_reqs.map(&:requirement).join("', '")}'`) " \
               "or revert to a stable version by running `pod update #{name}`."
+          elsif local_pod_parent
```

Comment:
There is some weird edge case, and I couldn't find a clean way to handle it. 

Suppose a local pod has dependency to pod A '1.8', and there is no '1.8' version of A in reality (in any of the pod sources).

If user runs `pod update A` in this case, installer will fail with the following:

```
[!] Unable to satisfy the following requirements:

- `A (= 1.8)` required by `LocalPod (1.0)`

It seems like you've changed the constraints of dependency `A` inside your development pod `LocalPod`.
You should run `pod update A` to apply changes you've made.
```

This is because here I print this warning message by checking if `requirement_trees` value includes a local pod.
We should also add some check which ensures that the given requirement actually exists in sources.

@segiddins what do you think is the best way to do this?